### PR TITLE
 User signUp controller done

### PR DESCRIPTION
**Pull request by java80**
Please review user controller sign up

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -5,3 +5,32 @@ const jwt = require("jsonwebtoken")
 
 
 db.on('error', console.error.bind(console, 'MongoDB connection error'))
+const SALT_ROUNDS = 11;
+const TOKEN_KEY = "thebestgroupever";
+const signUp = async (req, res) => {
```
Comment (wjb108) [inline controllers/users.js:10]: isn't the token key "securetoken" ?

File: controllers/users.js [hunk 2]
```diff
@@ -5,3 +5,32 @@ const jwt = require("jsonwebtoken")
...
+  // create jwt token
+  // send back token
+  try {
+    const { username, email, password } = req.body;
+    const password_digest = await bcrypt.hash(password, SALT_ROUNDS);
+    const user = new User({ username, email, password_digest });
+    await user.save();
+    const payload = {
+      username: user.username,
+
```
Comment (wjb108) [inline controllers/users.js:20]: i believe your misssing the email field in the payload

File: controllers/users.js [hunk 3]
```diff
@@ -5,3 +5,32 @@ const jwt = require("jsonwebtoken")
...
+    };
+    const token = jwt.sign(payload, TOKEN_KEY);
+    return res.status(201).json({ token, payload });
+  } catch (error) {
```
Comment (wjb108) [inline controllers/users.js:24]: in the quacking passwords, repo it only has on line 27 token in the {}

File: controllers/users.js (hunk 1 above)
Comment (java80) [reply controllers/users.js:10]: you guys can decide to change it

File: controllers/users.js (hunk 2 above)
Comment (java80) [reply controllers/users.js:20]: thanks will

File: controllers/users.js (hunk 1 above)
Comment (foremanlb) [reply controllers/users.js:10]: It has to be the same as the other token that is already established.

**Review by foremanlb — APPROVED** [review]

**Pull request merged**